.github/scripts/trymerge.py

```diff
@@ -816,42 +816,47 @@ def merge_on_green(pr_num: int, repo: GitRepo, dry_run: bool = False) -> None:
     org, project = repo.gh_owner_and_name()
     start_time = time.time()
     last_exception = ''
-    while True:
+    elapsed_time = 0.0
+    while elapsed_time < 400 * 60:
         current_time = time.time()
         elapsed_time = current_time - start_time
 
-        if(elapsed_time > 355 * 60):
-            msg = 'Merged timed out after 6 hours. Please contact the pytorch_dev_infra team.'
-            msg += f'The last exception was: {last_exception}'
-            gh_post_comment(org, project, pr_num, msg, dry_run=dry_run)
-            gh_add_labels(org, project, pr_num, ["land-failed"])
-            raise RuntimeError(msg)
 
         pr = GitHubPR(org, project, pr_num)
         try:
-            pr.merge_into(repo, dry_run=dry_run)
+            return pr.merge_into(repo, dry_run=dry_run)
         except MandatoryChecksMissingError as ex:
             last_exception = str(ex)
-            print(f'Merged failed due to: {ex}. Retrying in 60 seconds.')
+            print(f"Merged failed due to: {ex}. Retrying in 60 seconds.")
             time.sleep(60)
-        else:
-            return
+    # Finally report timeout back
+    msg = "Merged timed out after 6 hours. Please contact the pytorch_dev_infra team."
+    msg += f"The last exception was: {last_exception}"
+    gh_post_comment(org, project, pr_num, msg, dry_run=dry_run)
```

Comment:
We are already posting comment inside `handle_exception`, so it is not required here.

Reply:
Good point, removing. cc: @zengk95